plugin/src/main/java/org/jenkinsci/plugins/structs/SymbolLookup.java

```diff
@@ -152,5 +153,36 @@ public static SymbolLookup get() {
         return j.getInjector().getInstance(SymbolLookup.class);
     }
 
+    /**
+     * Get the {@link Symbol} value for the class of the given object, generally a {@link Descriptor}, if the annotation
+     * is present. If the object is in fact a {@link Describable}, we'll use its {@link Descriptor} class instead.
+     *
+     * @param o An object
+     * @return The {@link Symbol} annotation value for the class (generally a {@link Descriptor} that object represents,
+     * or null if the annotation is not present.
+     */
+    public static String getSymbolValue(Object o) {
+        if (o instanceof Describable) {
+            return getSymbolValue(((Describable) o).getDescriptor().getClass());
+        } else {
+            return getSymbolValue(o.getClass());
+        }
+    }
+
+    /**
+     * Get the {@link Symbol} value for the given class, if the annotation is present. Unlike {@link #getSymbolValue(Object)},
+     * this will not get the {@link Descriptor} for {@link Describable} classes.
```

Comment:
Yeah, I was debating whether I even *wanted* to do that - I came to the decision that I'd prefer to have the `Class` version *only* work if you give it an annotated class without the smarts behind it.